### PR TITLE
Don't request the freshest data, wait a bit.

### PR DESCRIPTION
**Pull request by brian-brazil**
Cloudwatch has been reported to take a few minutes
to converge on values, so as we don't have a way to update
old values wait a bit to be safe.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,16 +50,29 @@ aws_elb_request_count_sum{job="aws_elb",load_balancer_name="mylb",availability_z
 aws_elb_request_count_sum{job="aws_elb",load_balancer_name="myotherlb",availability_zone="eu-west-1c",} 7.0
 ```
 
-All metrics are exported as gauges. Timestamps from Cloudwatch are not passed to Prometheus, pending resolution of [#398](https://github.com/prometheus/prometheus/issues/398).
+All metrics are exported as gauges.
 
-In addition `cloudwatch_exporter_scrape_error` will be non-zero if an error occured during the scrape, and `cloudwatch_exporter_scrape_duration_seconds` has the duration of that scrape.
+Timestamps from Cloudwatch are not passed to Prometheus, pending resolution of
+[#398](https://github.com/prometheus/prometheus/issues/398). Clourwatch has
```
Comment (juliusv) [inline README.md]: s/Clourwatch/CloudWatch/

s/Cloudwatch/CloudWatch/

File: README.md [hunk 2]
```diff
@@ -45,16 +50,29 @@ aws_elb_request_count_sum{job="aws_elb",load_balancer_name="mylb",availability_z
...
+been observed to sometimes take minutes for reported values to converge. The
+default `delay_seconds` will result in data that is at least 5 minutes old
+being requested to mitigate this.
+
+In addition `cloudwatch_exporter_scrape_error` will be non-zero if an error
+occured during the scrape, and `cloudwatch_exporter_scrape_duration_seconds`
```
Comment (juliusv) [inline README.md]: s/occured/occurred/

File: README.md [hunk 3]
```diff
@@ -45,16 +50,29 @@ aws_elb_request_count_sum{job="aws_elb",load_balancer_name="mylb",availability_z
...
+has the duration of that scrape.
```
Comment (juliusv) [inline README.md]: s/has/contains/

or

s/has/exports/

File: README.md [hunk 4]
```diff
@@ -45,16 +50,29 @@ aws_elb_request_count_sum{job="aws_elb",load_balancer_name="mylb",availability_z
...
 
 ### Cost
 
 Amazon charges for every API request, see the [current charges](http://aws.amazon.com/cloudwatch/pricing/).
 
-Every metric retrieved requires one API request, which can include multiple statistics. In addition, when `aws_dimensions` is provided, the exporter needs to do API requests to determine what metrics to request. This should be negligable compared to the requests for the metrics themselvs.
+Every metric retrieved requires one API request, which can include multiple
+statistics. In addition, when `aws_dimensions` is provided, the exporter needs
+to do API requests to determine what metrics to request. This should be
+negligable compared to the requests for the metrics themselves.
```
Comment (juliusv) [inline README.md]: negligible

**Comment by brian-brazil** [comment]
Updated.

File: README.md [hunk 5]
```diff
@@ -11,7 +11,11 @@ An exporter for [Amazon Cloudwatch](http://aws.amazon.com/cloudwatch/), for Prom
 
 ## Credentials and permissions
 
-The Cloudwatch Exporter uses the [AWS Java SDK](http://docs.aws.amazon.com/AWSSdkDocsJava/latest/DeveloperGuide/welcome.html), which offers [a variety of ways to provide credentials](http://docs.aws.amazon.com/AWSSdkDocsJava/latest/DeveloperGuide/credentials.html). This includes the `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` enviroment variables
+The Cloudwatch Exporter uses the
```
Comment (juliusv) [inline README.md]: s/Cloudwatch/CloudWatch/ everywhere

Comment (brian-brazil) [inline README.md]: Okay, if that's the convention we're going to follow.

Comment (juliusv) [inline README.md]: It's just the spelling that Amazon uses, so it's good to spell their thing like they do :)

Comment (brian-brazil) [inline README.md]: That affects other things too, we've other exporters that have non-obvious capitalisation to those not used to them such as HAProxy.

Comment (juliusv) [inline README.md]: HAProxy is mostly spelled "HAProxy" by HAProxy: http://www.haproxy.org/

In any case, we don't have to spend extraordinary amounts of effort to
always use a particularly sanctioned spelling, but if it's easy enough to
do so (like now, when we have the occasion), we should.

**Comment by juliusv** [comment]
:+1:

**Comment by dev-e** [comment]
Hi! Please explain why
> we don't have a way to update old values
>
Is it possible to export the same metrics but with different timestamps? For example for 5 minutes back from scrape and update them.
e.g.:
`aws_applicationelb_request_count_sum{job="aws_applicationelb",instance="",load_balancer="app/lb/xxxxxxxxx",} 2191.0 1562748120000
aws_applicationelb_request_count_sum{job="aws_applicationelb",instance="",load_balancer="app/lb/xxxxxxxxx",} 3422.0 1562748180000
aws_applicationelb_request_count_sum{job="aws_applicationelb",instance="",load_balancer="app/lb/xxxxxxxxx",} 2322.0 1562748240000
aws_applicationelb_request_count_sum{job="aws_applicationelb",instance="",load_balancer="app/lb/xxxxxxxxx",} 1222.0 1562748300000
aws_applicationelb_request_count_sum{job="aws_applicationelb",instance="",load_balancer="app/lb/xxxxxxxxx",} 2350.0 1562748360000`

Next scrape (on new minute)
`aws_applicationelb_request_count_sum{job="aws_applicationelb",instance="",load_balancer="app/lb/xxxxxxxxx",} 3422.0 1562748180000
aws_applicationelb_request_count_sum{job="aws_applicationelb",instance="",load_balancer="app/lb/xxxxxxxxx",} 2322.0 1562748240000
aws_applicationelb_request_count_sum{job="aws_applicationelb",instance="",load_balancer="app/lb/xxxxxxxxx",} 3222.0 1562748300000
aws_applicationelb_request_count_sum{job="aws_applicationelb",instance="",load_balancer="app/lb/xxxxxxxxx",} 3760.0 1562748360000
aws_applicationelb_request_count_sum{job="aws_applicationelb",instance="",load_balancer="app/lb/xxxxxxxxx",} 235.0 1562748420000`

**Pull request merged**